### ISSUE TITLE
4.1.6: Frequent Exception caught @ class io.netty.channel.unix.Errors$NativeIoException : syscall:read(...)() failed: Connection reset by peer

### ISSUE THREAD

**Issue by fangqiao** [netty/netty#6266]
I am using 4.1.6 as REST API service infrastructure. 

During testing, Netty often reports this exception:

Exception caught @ class io.netty.channel.unix.Errors$NativeIoException : syscall:read(...)() failed: Connection reset by peer.

And the client current REST request seems to receive the response of an earlier REST request that has been returned successfully when this exception happens.

This happens quite frequently.  I tried it on 4.1.7. It happens as well.

I use java 1.8.0_111. Server is centos 7.

Also Nginx is used as reverse proxy to connect to netty that runs plain HTTP.

**Comment by Scottmitch** [comment]
This means your peer has sent a `RST`. Can you describe what the Netty issue is?

**Comment by fangqiao** [comment]
There was an old issue [3539](https://github.com/netty/netty/issues/3539) of similar problem. I thought the problem might have resurfaced. On the other hand, it might be the Nginx reverse proxy configuration problem. I am looking into it.

Thanks

**Comment by Scottmitch** [comment]
As far as I know the issue you referenced should still be resolved. If you are able to provide a reproducer that would help us track down the issue.

**Comment by qierangtianxia** [comment]
ok,i just look

**Comment by fangqiao** [comment]
I looked in to the problem. It looks it is indeed a nginx issue.

My nginx reverse proxy server is configured with two upstream REST service servers based o Netty. For convenience of testing, I shut down one instance of the REST service server. However the Nginx upstream still was configured with two upstream servers. 

It seems it is this set up that caused the issue. After I turned the second REST service on, the problem disappeared. 

I am not sure if it is a bug for Nginx. I think even under this setup, Nginx shouldn't prematurely close connection with upstream server that it has connected to.
